proof/refine/RISCV64/VSpace_R.thy

```diff
@@ -194,58 +173,28 @@ lemma hwASIDFlush_corres[corres]:
   "corres dc \<top> \<top> (do_machine_op (hwASIDFlush x)) (doMachineOp (hwASIDFlush x))"
   by (corresKsimp corres: corres_machine_op)
 
-lemma deleteASID_corres [corres]:
+lemma deleteASID_corres[corres]:
   assumes "asid' = ucast asid" "pm' = pm"
   shows "corres dc invs no_0_obj'
                 (delete_asid asid pm) (deleteASID asid' pm')"
   unfolding delete_asid_def deleteASID_def using assms
   apply simp
-  apply (rule corres_guard_imp)
-    apply (rule corres_split[OF corres_gets_asid])
-      apply (case_tac "asid_table (asid_high_bits_of asid)", simp)
-      apply clarsimp
-      apply (rule_tac P="\<lambda>s. asid_high_bits_of asid \<in> dom (asidTable o ucast) \<longrightarrow>
-                             asid_pool_at (the ((asidTable o ucast) (asid_high_bits_of asid))) s \<and>
-                             pspace_aligned s \<and> pspace_distinct s" and
-                      P'="\<top>" and
-                      Q="invs and
-                         (\<lambda>s. asid_table s = asidTable \<circ> ucast)" in
-                      corres_split)
-         apply (simp add: dom_def)
-         apply (rule get_asid_pool_corres_inv'[OF refl, unfolded pred_conj_def, simplified])
-        apply (rule corres_when)
+  apply corres_pre
+    apply (corres_split, corres)
+      apply corres_cases_both
```

Comment:
I'd enjoy a comment here saying which cases these are (so can tell where corres got to without looking at the proof state.
Otherwise, looks like the `corres` method and its cousins are very useful indeed.